### PR TITLE
WIP - Author and Game class

### PR DESCRIPTION
**Pull request by muneebulrehman**
# Added basic structure of Author and Game class according to UML diagram
- Added necessary variables
- Added necessary functions

### CODE REVIEW

**Comment by ahangarha** [comment]
I am going to review this

File: lib/game.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Game
+  attr_accessor :multiplayer, :last_played_at
+
+  def initialize(multiplayer, last_played_at)
+    @multiplayer = multiplayer
+    @last_played_at = last_played_at
+  end
+end
```
Comment (ahangarha) [inline lib/game.rb:8]: As an independent class, this is a correct implementation of UML diagram. But please consider that Game class has aggregation (inheritance) relationship with Item class. This means when we make an instance of Game class, we are actually creating an Item.

File: lib/author.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  def initialize(first_name, last_name)
+    @first_name = first_name
+    @last_name = last_name
+    @id = rand(1..1000)
+    @items = []
+  end
+
+  def add_item(item)
+    @items << item unless @items.include?(item)
+  end
```
Comment (ahangarha) [inline lib/author.rb:14]: Good job. Please consider the other side of the relationship: the Item. If we add an Item to an Author, the Author should be added to the associated Item as well.

**Review by ahangarha — CHANGES_REQUESTED** [review]
Good job so far. Very fast.

## CHANGES REQUIRED :no_entry_sign: 



I have already added comments for relevant parts of the committed code. Please take a look at them and make the changes.

**Comment by muneebulrehman** [comment]
@ahangarha, as I mentioned in the title, this is a basic structure. Please keep patience. I can't add everything in one go. This is a slow process. These things will be added as the project progresses.

**Comment by ahangarha** [comment]
Ok! Since you shared the link in Slack I though you need review.

**Comment by muneebulrehman** [comment]
> Ok! Since you shared the link in Slack I though you need review.

Nope, there wasn't much to review. That is why I mentioned in the title as a basic structure

**Pull request closed without merging**